packages/flutter/test/widgets/framework_test.dart

```diff
@@ -1574,6 +1574,23 @@ void main() {
     expect(RawKeyboard.instance.keyEventHandler, same(rawKeyEventHandler));
   });
 
+  testWidgets('Incorrect use of ParentDataWidget shows a useful error message', (WidgetTester tester) async {
+    await tester.pumpWidget(
+      Container(
+        key: const Key('container'),
+        child: Flexible(
+          child: Container(),
+        ) // Flexible should not be used directly in a Container or other inappropriate parents
+      ),
+    );
+
+    await tester.pumpAndSettle();
+
+    final dynamic exception = tester.takeException();
+
+    expect(exception, isFlutterError);
```

Comment:
I would prevent the use of `dynamic` here. (If you really want a type, use `Object?` instead)

```suggestion
    expect(tester.takeException(), isFlutterError);
```